File: src/utils/seed.js
import { PrismaClient } from '@prisma/client';
import bcrypt from 'bcryptjs';

const prisma = new PrismaClient();

async function main() {
  console.log('🌱 Seeding database...');

  // Sports
  const sports = await Promise.all([
    prisma.sport.upsert({ where: { slug: 'football' }, update: {}, create: { name: 'Football', slug: 'football', icon: '⚽' } }),
    prisma.sport.upsert({ where: { slug: 'basketball' }, update: {}, create: { name: 'Basketball', slug: 'basketball', icon: '🏀' } }),
    prisma.sport.upsert({ where: { slug: 'tennis' }, update: {}, create: { name: 'Tennis', slug: 'tennis', icon: '🎾' } }),
    prisma.sport.upsert({ where: { slug: 'cricket' }, update: {}, create: { name: 'Cricket', slug: 'cricket', icon: '🏏' } }),
    prisma.sport.upsert({ where: { slug: 'boxing' }, update: {}, create: { name: 'Boxing', slug: 'boxing', icon: '🥊' } }),
  ]);

  const football = sports[0];
  const basketball = sports[1];

  // Events
  const now = new Date();
  const events = [
    {
      sportId: football.id, homeTeam: 'Manchester City', awayTeam: 'Arsenal',
      startTime: new Date(now.getTime() + 2 * 60 * 60 * 1000),
      status: 'UPCOMING', league: 'Premier League', country: 'England',
    },
    {
      sportId: football.id, homeTeam: 'Real Madrid', awayTeam: 'Barcelona',
      startTime: new Date(now.getTime() + 4 * 60 * 60 * 1000),
      status: 'UPCOMING', league: 'La Liga', country: 'Spain',
    },
    {
      sportId: football.id, homeTeam: 'PSG', awayTeam: 'Bayern Munich',
      startTime: new Date(now.getTime() - 30 * 60 * 1000),
      status: 'LIVE', league: 'Champions League', homeScore: 1, awayScore: 0,
    },
    {
      sportId: basketball.id, homeTeam: 'LA Lakers', awayTeam: 'Golden State Warriors',
      startTime: new Date(now.getTime() + 6 * 60 * 60 * 1000),
      status: 'UPCOMING', league: 'NBA', country: 'USA',
    },
    {
      sportId: basketball.id, homeTeam: 'Boston Celtics', awayTeam: 'Miami Heat',
      startTime: new Date(now.getTime() - 45 * 60 * 1000),
      status: 'LIVE', league: 'NBA', homeScore: 67, awayScore: 54,
    },
  ];

  for (const eventData of events) {
    const event = await prisma.event.create({ data: eventData });

    // Create odds for each event
    if (event.sportId === football.id) {
      await prisma.odd.createMany({
        data: [
          { eventId: event.id, market: '1X2', selection: 'Home', value: +(1.5 + Math.random()).toFixed(2) },
          { eventId: event.id, market: '1X2', selection: 'Draw', value: +(2.8 + Math.random() * 0.5).toFixed(2) },
          { eventId: event.id, market: '1X2', selection: 'Away', value: +(2.5 + Math.random() * 1.5).toFixed(2) },
          { eventId: event.id, market: 'BTTS', selection: 'Yes', value: +(1.6 + Math.random() * 0.4).toFixed(2) },
          { eventId: event.id, market: 'BTTS', selection: 'No', value: +(2.0 + Math.random() * 0.5).toFixed(2) },
          { eventId: event.id, market: 'Over/Under', selection: 'Over 2.5', value: +(1.8 + Math.random() * 0.4).toFixed(2) },
          { eventId: event.id, market: 'Over/Under', selection: 'Under 2.5', value: +(1.9 + Math.random() * 0.4).toFixed(2) },
        ]
      });
    } else {
      await prisma.odd.createMany({
        data: [
          { eventId: event.id, market: 'Moneyline', selection: 'Home', value: +(1.5 + Math.random()).toFixed(2) },
          { eventId: event.id, market: 'Moneyline', selection: 'Away', value: +(2.0 + Math.random()).toFixed(2) },
          { eventId: event.id, market: 'Over/Under', selection: 'Over 210.5', value: +(1.85 + Math.random() * 0.2).toFixed(2) },
          { eventId: event.id, market: 'Over/Under', selection: 'Under 210.5', value: +(1.85 + Math.random() * 0.2).toFixed(2) },
        ]
      });
    }
  }

  // Demo user
  const password = await bcrypt.hash('password123', 12);
  await prisma.user.upsert({
    where: { email: 'demo@strikeodds.com' },
    update: {},
    create: {
      email: 'demo@strikeodds.com',
      username: 'demo_user',
      password,
      firstName: 'Demo',
      lastName: 'User',
      balance: 1000,
      transactions: { create: { type: 'DEPOSIT', amount: 1000, reference: 'Welcome bonus' } }
    }
  });

  console.log('✅ Seed complete!');
  console.log('📧 Demo login: demo@strikeodds.com / password123');
}

main().catch(console.error).finally(() => prisma.$disconnect());

File: seed.py
import sys, os
sys.path.insert(0, os.path.dirname(os.path.dirname(__file__)))

from datetime import datetime, timezone, timedelta
from app import create_app
from app.models import db, User, Sport, Event, Odd, Transaction
import bcrypt

app = create_app()

with app.app_context():
    db.create_all()

    # Sports
    sports_data = [
        ("Football", "football", "⚽"),
        ("Basketball", "basketball", "🏀"),
        ("Tennis", "tennis", "🎾"),
        ("Cricket", "cricket", "🏏"),
        ("Boxing", "boxing", "🥊"),
    ]
    sports = {}
    for name, slug, icon in sports_data:
        s = Sport.query.filter_by(slug=slug).first()
        if not s:
            s = Sport(name=name, slug=slug, icon=icon)
            db.session.add(s)
            db.session.flush()
        sports[slug] = s

    now = datetime.now(timezone.utc)

    # Events
    events_data = [
        {"sport": "football", "home": "Manchester City", "away": "Arsenal",      "time": now + timedelta(hours=2),  "status": "UPCOMING", "league": "Premier League"},
        {"sport": "football", "home": "Real Madrid",      "away": "Barcelona",   "time": now + timedelta(hours=4),  "status": "UPCOMING", "league": "La Liga"},
        {"sport": "football", "home": "PSG",              "away": "Bayern",      "time": now - timedelta(minutes=30), "status": "LIVE",   "league": "Champions League", "hs": 1, "as": 0},
        {"sport": "basketball","home": "LA Lakers",       "away": "Golden State","time": now + timedelta(hours=6),  "status": "UPCOMING", "league": "NBA"},
        {"sport": "basketball","home": "Boston Celtics",  "away": "Miami Heat",  "time": now - timedelta(minutes=45), "status": "LIVE",   "league": "NBA", "hs": 67, "as": 54},
    ]

    for ed in events_data:
        e = Event(
            sport_id=sports[ed["sport"]].id,
            home_team=ed["home"], away_team=ed["away"],
            start_time=ed["time"].replace(tzinfo=None),
            status=ed["status"], league=ed["league"],
            home_score=ed.get("hs"), away_score=ed.get("as")
        )
        db.session.add(e)
        db.session.flush()

        if ed["sport"] == "football":
            odds = [
                ("1X2", "Home", 1.75), ("1X2", "Draw", 3.20), ("1X2", "Away", 4.50),
                ("BTTS", "Yes", 1.65), ("BTTS", "No", 2.10),
                ("Over/Under", "Over 2.5", 1.85), ("Over/Under", "Under 2.5", 1.90),
            ]
        else:
            odds = [
                ("Moneyline", "Home", 1.60), ("Moneyline", "Away", 2.30),
                ("Over/Under", "Over 210.5", 1.87), ("Over/Under", "Under 210.5", 1.87),
            ]

        for market, selection, value in odds:
            db.session.add(Odd(event_id=e.id, market=market, selection=selection, value=value))

    # Demo user
    if not User.query.filter_by(email="demo@strikeodds.com").first():
        hashed = bcrypt.hashpw(b"password123", bcrypt.gensalt()).decode()
        user = User(
            email="demo@strikeodds.com", username="demo_user",
            password=hashed, first_name="Demo", last_name="User", balance=1000
        )
        db.session.add(user)
        db.session.flush()
        db.session.add(Transaction(user_id=user.id, type="DEPOSIT", amount=1000, reference="Welcome bonus"))

    db.session.commit()
    print("✅ Database seeded!")
    print("📧 Demo login: demo@strikeodds.com / password123")
